service/service_construction.go

```diff
@@ -225,6 +227,15 @@ func (s ConstructionService) ConstructionDerive(
 	ctx context.Context,
 	req *types.ConstructionDeriveRequest,
 ) (*types.ConstructionDeriveResponse, *types.Error) {
+	if mapper.IsPChain(req.NetworkIdentifier) {
```

Comment:
You should put this under the `req.PublicKey == nil` check.

Reply:
done :)